src/text/writer.rs

```diff
@@ -0,0 +1,352 @@
+use crate::result::{illegal_operation, IonResult};
+use crate::text::encoder::TextEncoder;
+use crate::IonType;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, FixedOffset};
+use std::io::{BufWriter, Write};
+
+pub struct TextWriter<W: Write> {
+    output: BufWriter<W>,
+    annotations: Vec<String>,
+    field_name: Option<String>,
+    containers: Vec<IonType>,
+}
+
+impl<W: Write> TextWriter<W> {
+    /// Constructs a new instance of TextWriter that writes values to the provided io::Write
+    /// implementation.
+    pub fn new(sink: W) -> TextWriter<W> {
+        TextWriter {
+            output: BufWriter::new(sink),
+            annotations: vec![],
+            field_name: None,
+            containers: vec![],
+        }
+    }
+
+    /// Returns a reference to the underlying io::Write implementation.
+    pub fn output(&self) -> &W {
+        self.output.get_ref()
+    }
+
+    /// Causes any buffered data to be written to the underlying io::Write implementation.
+    pub fn flush(&mut self) -> IonResult<()> {
+        self.output.flush()?;
+        Ok(())
+    }
```

Comment:
In other APIs, we have a difference of `flush` vs. `finish`, do you have thoughts about that here?

Reply:
The strongest use case I'm aware of for having both is LST appends. Since the text writer doesn't maintain an LST itself and we don't yet have a `Writer` trait to code to, I'm for punting on this.